game-core/src/main/java/games/strategy/triplea/ui/unit/scroller/UnitScroller.java

```diff
@@ -0,0 +1,278 @@
+    panel.add(Box.createVerticalStrut(2));
+
+    territoryNameLabel.setAlignmentX(JComponent.CENTER_ALIGNMENT);
+    panel.add(territoryNameLabel);
+
+    final JPanel centerPanel = new JPanel();
+    centerPanel.setLayout(new BoxLayout(centerPanel, BoxLayout.X_AXIS));
+
+    final JButton prevUnit = new JButton(UnitScrollerIcon.LEFT_ARROW.get());
+    prevUnit.setToolTipText(PREVIOUS_UNITS_TOOLTIP);
+    prevUnit.setFocusable(false);
+    prevUnit.setAlignmentX(JComponent.CENTER_ALIGNMENT);
+    prevUnit.addActionListener(e -> centerOnPreviousMovableUnit());
+    centerPanel.add(prevUnit);
+    centerPanel.add(selectUnitImagePanel);
+
+    final JButton nextUnit = new JButton(UnitScrollerIcon.RIGHT_ARROW.get());
+    nextUnit.setToolTipText(NEXT_UNITS_TOOLTIP);
+    nextUnit.setFocusable(false);
+    nextUnit.addActionListener(e -> centerOnNextMovableUnit());
+    centerPanel.add(nextUnit, BorderLayout.EAST);
+    centerPanel.add(Box.createHorizontalStrut(10));
+
+    panel.add(centerPanel);
+
+    final JButton skipButton = new JButton(UnitScrollerIcon.UNIT_SKIP.get());
+    skipButton.setToolTipText(SKIP_UNITS_TOOLTIP);
+    skipButton.setFocusable(false);
+    skipButton.setAlignmentX(JComponent.CENTER_ALIGNMENT);
+    skipButton.addActionListener(e -> skipCurrentUnits());
+    panel.add(skipButton, BorderLayout.SOUTH);
+    panel.add(Box.createVerticalStrut(3));
+
+    panel.add(SwingComponents.leftBox(movesLeftLabel));
+
+    panel.add(Box.createVerticalStrut(5));
+    return panel;
+  }
+
+  /** Centers the map on the current territory shown in the unit scroller. */
+  public void centerOnCurrentMovableUnit() {
+    if (lastFocusedTerritory != null) {
+      mapPanel.setUnitHighlight(
+          Collections.singleton(
+              UnitScrollerModel.getMoveableUnits(
+                  lastFocusedTerritory,
+                  movePhaseSupplier.get(),
+                  currentPlayerSupplier.get(),
+                  skippedUnits)));
+      mapPanel.centerOnTerritoryIgnoringMapLock(lastFocusedTerritory);
+    } else {
+      centerOnNextMovableUnit();
+    }
+  }
+
+  /**
+   * Skips the units in the current territory. Scroller will not scroll back to these units and the
+   * units will not be highlighted.
+   */
+  public void skipCurrentUnits() {
+    if (lastFocusedTerritory != null) {
+      skippedUnits.addAll(lastFocusedTerritory.getUnits());
+      updateMovesLeftLabel();
+    }
+    centerOnNextMovableUnit();
+  }
+
+  public void centerOnNextMovableUnit() {
+    centerOnMovableUnit(true);
+  }
+
+  public void centerOnPreviousMovableUnit() {
+    centerOnMovableUnit(false);
+  }
+
+  private void centerOnMovableUnit(final boolean selectNext) {
+    final List<Territory> allTerritories;
+    gameData.acquireReadLock();
+    try {
+      allTerritories = new ArrayList<>(gameData.getMap().getTerritories());
+    } finally {
+      gameData.releaseReadLock();
+    }
+
+    final int size = allTerritories.size();
```

Comment:
I know this code is copied over, but this intermediate variable is redundant.
`allTerritories` is never modified in the code below.

Reply:
Agree.

The read lock is not useful either. If the territory list is modified, the code below is wrong anyways. We do not expect the territory list to be modified while it is the current players turn anyways, it's just about the only way for territories to be modified, so it's an impossible situation.

I removed the read lock and the extra list copies and the `size` intermediate variable in: 877f9b453